Source/WebCore/platform/graphics/texmap/TextureMapperLayer3DRenderingContext.cpp

```diff
@@ -75,72 +118,175 @@ void TextureMapperLayer3DRenderingContext::paint(TextureMapper& textureMapper, c
     clipBuffer->updateData(clipVertices.data(), 0, clipBufferSize);
 
     // Paint
-    traverseTree(*root, [&clipVertices, &clipBuffer, &paintLayerFunction](TextureMapperLayerNode& node) {
-        for (auto& polygon : node.polygons) {
-            unsigned numberOfClipVertices = polygon.isSplitted ? polygon.geometry.numberOfVertices() : 0;
+    traverseTree(*root, [&clipVertices, &clipBuffer, &paintLayerFunction](LayerNode& node) {
+        for (auto& layer : node.layers) {
+            unsigned numberOfClipVertices = layer.isSplitted ? layer.geometry.numberOfVertices() : 0;
 
             Vector<FloatPoint> points;
             if (numberOfClipVertices > 0) {
                 points.reserveCapacity(numberOfClipVertices);
-                auto xy = clipVertices.subvector(polygon.clipVertexBufferOffset, numberOfClipVertices * 2);
+                auto xy = clipVertices.subvector(layer.clipVertexBufferOffset, numberOfClipVertices * 2);
                 for (size_t i = 0; i < xy.size(); i += 2)
                     points.append(FloatPoint(xy.at(i), xy.at(i + 1)));
             }
 
-            ClipPath clipPath(WTFMove(points), clipBuffer->bufferID(), polygon.clipVertexBufferOffset * sizeof(float));
+            ClipPath clipPath(WTFMove(points), clipBuffer->bufferID(), layer.clipVertexBufferOffset * sizeof(float));
 
-            paintLayerFunction(polygon.layer, clipPath);
+            paintLayerFunction(layer.textureMapperLayer, clipPath);
         }
     });
 }
 
-// Build BSP tree for rendering polygons with painter's algorithm.
+TextureMapperLayer3DRenderingContext::SweepAndPrunePairs TextureMapperLayer3DRenderingContext::sweepAndPruneXAxis(const Vector<Layer>& layers)
+{
+    auto layerBoundsXRange = [](const Layer& layer) -> std::pair<float, float> {
```

Comment:
This could be an helpful accessor for Layer directly?

Reply:
not needed anymore